pkg/openshift/certgen/config_test.go

```diff
@@ -0,0 +1,116 @@
+package certgen
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/Azure/acs-engine/pkg/openshift/filesystem"
+)
+
+type fakefilesystem struct {
+}
+
+func (fakefilesystem) WriteFile(filename string, data []byte, perm os.FileMode) error {
+	// check that you've been asked to write the given file with sane permissions & owner
+	fmt.Printf("Filename: %v	Permissions: %04o   string: %v\n", filename, perm.Perm(), perm.String())
```

Comment:
the fakefilesystem needs to stash somewhere details about what it's been asked to write so that you can inspect it at the end of your test